### PR TITLE
jpeg: Move component esp_jpeg from BSP and add test.

### PR DESCRIPTION
**Pull request by espzav**
# Checklist

- [x] Component contains License
- [x] Component contains README.md
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [x] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [x] _Optional:_ Component contains unit tests
- [ ] CI passing

### CODE REVIEW

**Comment by espzav** [comment]
@suda-morris PTAL

File: esp_jpeg/test/tjpgd_test.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <string.h>
```
Comment (tore-espressif) [inline esp_jpeg/test/tjpgd_test.c:2]: Please add license + author header

**Review by tore-espressif — APPROVED** [review]
@espzav thank you for the follow-up. 

Are there any notable changes from the version in esp=bsp? (v1.0.2)

File: esp_jpeg/test/tjpgd_test.c (hunk 1 above)
Comment (espzav) [reply esp_jpeg/test/tjpgd_test.c:7]: No change in esp_jpeg component. Only added test and moved from BSP. I increased the version for upload changed URL to component manager.

**Pull request merged**